### PR TITLE
[RORDEV-1079] fix: Repositories and Snapshots handling by ES coordinating nodes

### PR DESCRIPTION
**Pull request by coutoPL**
🐞Fix (ES) [Repositories and Snapshots handling by ES coordinating nodes](https://forum.readonlyrest.com/t/snapshot-status-cannot-modify-incoming-request/2471)

### CODE REVIEW

**Review by mateuszkp96 — CHANGES_REQUESTED** [review]

File: es65x/src/main/scala/tech/beshu/ror/es/utils/EsPatchVerifier.scala [hunk 1]
```diff
@@ -16,54 +16,37 @@
...
+    val result = for {
+      esHome <- pathHomeFrom(settings)
+      esPatch <- createPatcher(esHome)
+    } yield {
+      esPatch.isPatched match {
+        case Yes =>
+        case No(Cause.PatchedWithDifferentVersion(expectedRorVersion, patchedByRorVersion)) =>
+          throw new IllegalStateException(s"Elasticsearch was patched using ROR $patchedByRorVersion patcher. It should be unpatched and patched again with current ROR patcher ($expectedRorVersion). ReadonlyREST cannot be started. For patching instructions see our docs: https://docs.readonlyrest.com/elasticsearch#3.-patch-elasticsearch")
+        case No(Cause.NotPatchedAtAll) =>
+          throw new IllegalStateException("Elasticsearch is not patched. ReadonlyREST cannot be started. For patching instructions see our docs: https://docs.readonlyrest.com/elasticsearch#3.-patch-elasticsearch")
```
[the remark this answers is not in the document]
Comment (mk-software-pl) [reply es65x/src/main/scala/tech/beshu/ror/es/utils/EsPatchVerifier.scala:40]: there is no cats-core in the ror-tools, so I didn't use Show typeclass. I think there is no need to add it now.

[the remark this answers is not in the document]
Comment (coutoPL) [reply es65x/src/main/scala/tech/beshu/ror/es/utils/EsPatchVerifier.scala:40]: there is not cats-core in the ror-tools so I didn't use Show typeclass. IMO there is no sense to add the dependency

File: es63x/src/main/scala/tech/beshu/ror/es/utils/EsPatchVerifier.scala [hunk 2]
```diff
@@ -61,7 +50,10 @@ object EsPatchVerifier extends Logging {
     for {
       esHome <- pathHomeFrom(settings)
       esPatch <- createPatcher(esHome)
-    } yield esPatch.isPatched
+    } yield esPatch.isPatched match {
```
[the remark this answers is not in the document]
Comment (coutoPL) [reply es63x/src/main/scala/tech/beshu/ror/es/utils/EsPatchVerifier.scala:53]: I removed the patching step from the `es63x` module. It's too problematic to support and I think in the near future we will remove the following modules: `es60x`, `es61x`, `es62x` & `es63x`

**Review by mateuszkp96 — APPROVED** [review]
LGTM

**Pull request merged**